onebusaway-android/src/main/java/org/onebusaway/android/tripservice/PollerTask.java

```diff
@@ -125,18 +127,19 @@ private void poll1(Cursor c) {
         ObaArrivalInfoResponse response = ObaArrivalInfoRequest
                 .newRequest(mContext, stopId).call();
 
-        Long departMS = null;
+        // Pair of eta and is arriving information
+        Pair<Long, Boolean> eta = null;
         if (response.getCode() == ObaApi.OBA_OK) {
-            departMS = checkArrivals(response, c);
+            eta = checkArrivals(response, c);
         }
 
-        if (departMS != null) {
-            final long diffTime = departMS - System.currentTimeMillis();
+        if (eta != null) {
+            final long diffTime = eta.first - System.currentTimeMillis();
```

Comment:
Instead of `System.currentTimeMillis()`, let's use `response.getCurrentTime()`.  This helps us move towards #29 and using current server time instead of device local time to avoid clock sync issues.  Please reference #29 in the commit description as well, so we can keep track of this progress.  This won't fix that issue entirely but will help move us in that direction.